Sources/Task/ExistentialTask.swift

```diff
@@ -57,15 +62,15 @@ public final class Task<SuccessValue>: NSObject {
     }
 
     /// Creates a task whose `upon(_:execute:)` methods use those of `base`.
-    public convenience init<Task: FutureProtocol>(_ base: Task, progress: Progress)
-        where Task.Value: Either, Task.Value.Left == Error, Task.Value.Right == SuccessValue {
-        self.init(future: Future(task: base), progress: progress)
+    public convenience init<OtherTask: TaskProtocol>(_ base: OtherTask, progress: Progress) where OtherTask.SuccessValue == SuccessValue {
```

Comment:
These go back and forth a few times in the project; not sure if the generic parameter should be `OtherTask` or just `Other`.